lib/utils/templatize.js

```diff
@@ -107,6 +107,48 @@ const templateInstanceBase = PropertyEffects(
     // "class {}" is a constructor function.
     /** @type {function(new:Object)} */(class {}));
 
+export function showHideChildren(hide, children) {
+  for (let i=0; i<children.length; i++) {
+    let n = children[i];
+    // Ignore non-changes
+    if (Boolean(hide) != Boolean(n.__hideTemplateChildren__)) {
+      // clear and restore text
+      if (n.nodeType === Node.TEXT_NODE) {
+        if (hide) {
+          n.__polymerTextContent__ = n.textContent;
+          n.textContent = '';
+        } else {
+          n.textContent = n.__polymerTextContent__;
+        }
+      // remove and replace slot
+      } else if (n.localName === 'slot') {
+        if (hide) {
+          n.__polymerReplaced__ = document.createComment('hidden-slot');
```

Comment:
Bookmark. This seems kinda of nuts v. just doing the show/hide logic for the assignedNodes. Remember why we did it this way?

Reply:
Doing that could get out of sync so let's leave this as is.